File: ci-demo-business/src/main/java/nc/indus/ci/demo/mapper/DogMapper.java
package nc.indus.ci.demo.mapper;

import nc.indus.ci.demo.dto.Dog;
import nc.indus.ci.demo.entities.DogEntity;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper
public interface DogMapper {

	@Mapping(source = "name", target = "name")
	@Mapping(source = "color", target = "color")
	Dog toDto(DogEntity dogEntity);
}

File: ci-demo-business/src/main/java/nc/indus/ci/demo/mapper/DogMapper.kt
package nc.indus.ci.demo.mapper

import nc.indus.ci.demo.dto.Dog
import nc.indus.ci.demo.entities.DogEntity
import org.mapstruct.Mapper
import org.mapstruct.Mapping

@Mapper
interface DogMapper {

    @Mapping(source = "name", target = "name")
    fun toDto(dogEntity: DogEntity): Dog
}
